cli/types/cypress.d.ts

```diff
@@ -10,10 +10,10 @@ declare namespace Cypress {
   type PrevSubject = keyof PrevSubjectMap
   type TestingType = 'e2e' | 'component'
   type PluginConfig = (on: PluginEvents, config: PluginConfigOptions) => void | ConfigOptions | Promise<ConfigOptions>
-
+	type JQuerySelector = JQuery & { selector: string }
```

Comment:
@atarek12 Looking at [this documentation](https://api.jquery.com/selector/#selector1), it appears the `.selector` internal property was deprecated in jQuery 1.7 and removed in 3.0. 

Cypress ships jquery `v3.1.1` so ti seems the current types are accurately reflecting what jquery supports.

Reply:
I see that, but when logging the element content, you will find it exists with the correct used `selector`

```ts
Cypress.Commands.add( "command", { prevSubject: "element" }, (element) => {
 console.log(element) // selector property found -> {selector: ".class"}
}

cy.get(".class").command()
```

Reply:
There are 2 things to note for me:

1\.

This `selector` is added by Cypress for error messages. 

https://github.com/cypress-io/cypress/blob/b031803cfb31594054e9797719e2dd3365d43fa1/packages/driver/src/cy/commands/querying/querying.ts#L284-L290

I'm not sure if it's OK to make it public, because Cypress brought it back for internal use. 

2\.

If we decide to bring it back, I think it should be typed with `interface` like below:

```ts
interface JQueryWithSelector<TElement = HTMLElement> extends JQuery<TElement> {
  selector?: string
}
```

Because we'll lose the more specific type info.

Reply:
for the first point, I stuck in a problem and luckily found the `selector` property that solved it. So, it may be useful for others too.

I totally agree with the second point and will add it.